src/log.rs

```diff
@@ -55,12 +67,50 @@ impl<T: Transport> Stream for LogStream<T> {
             #[project]
             let next_state = match state.as_mut().project() {
                 LogStreamState::CreatingFilter(create_filter, poll_interval) => {
-                    let log_filter = match ready!(create_filter.poll(cx)) {
-                        Ok(log_filter) => log_filter,
+                    let filter = match ready!(create_filter.poll(cx)) {
+                        Ok(filter) => filter,
                         Err(err) => return Poll::Ready(Some(Err(err.into()))),
                     };
-                    let stream = log_filter.stream(*poll_interval).compat();
-                    LogStreamState::Streaming(stream)
+                    // TODO: this request can be very slow and the response very big. We might want
+                    // to be smarter about it. However, there is no way to page this request. Maybe
+                    // Set up intermediate queries to handle a limited number of blocks at a time.
+                    let past_logs_future = CompatCallFuture::<T, Vec<Log>>::new(filter.logs());
+                    LogStreamState::GettingPastLogs {
```

Comment:
I think that we shouldn't enter this state unconditionally. I think that if `from_block` is `latest` or `pending` we can skip straight to streaming.